File: programs-basic-java/src/main/java/com/local/numbertypes/LongValue.java
package com.local.numbertypes;

import static java.lang.Long.valueOf;

import java.util.function.Predicate;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class LongValue implements Predicate<String> {

	@Override
	public boolean test(String line) {
		try {
			valueOf(line);
		} catch (NumberFormatException exception) {
			log.debug("Long number invalid: {}", line);
			return Boolean.FALSE;
		}
		return Boolean.TRUE;
	}
}

File: programs-basic-kotlin/src/main/kotlin/com/local/numbertypes/LongValue.kt
package com.local.numbertypes

import mu.KotlinLogging
import java.util.function.Predicate

private val logger = KotlinLogging.logger {}
class LongValue : Predicate<String> {
	override fun test(line: String): Boolean {
		return try {
			line.toLong();
			true;
		} catch (exception: NumberFormatException) {
			logger.debug("Long number invalid: {}", line)
			false
		}
	}
}